sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/models/ModelMetadata.java

```diff
@@ -0,0 +1,40 @@
+package com.azure.iot.modelsrepository.implementation.models;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+
+/**
+ * ModelMetadata is designated to store KPIs from model parsing.
+ */
+public class ModelMetadata {
+    private String id;
+    private ArrayList<String> extend;
+    private ArrayList<String> componentSchemas;
+    private LinkedHashSet<String> dependencies;
+
+    public ModelMetadata(String id, ArrayList<String> extend, ArrayList<String> componentSchemas) {
+        this.id = id;
+        this.extend = extend;
+        this.componentSchemas = componentSchemas;
+        this.dependencies = new LinkedHashSet<>(extend);
+        this.dependencies.addAll(componentSchemas);
+    }
+
+    public String getId() {
```

Comment:
Lombok has a useful annotation that can replace these simple getters if you are interested. [Here's a good example](https://github.com/Azure/azure-iot-sdk-java/blob/f3204504532d07a5d907fabf485402c6896ffb57/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClientOptions.java#L17)

Reply:
I don't think we can take a dependency to lombok unfortunately :(